aten/src/ATen/native/BatchLinearAlgebra.cpp

```diff
@@ -442,6 +488,40 @@ TORCH_META_FUNC(linalg_lu_factor_ex)(const Tensor& A, bool pivot, bool check_err
   set_output(2, sizes, {}, A.options().dtype(kInt), {});
 }
 
+TORCH_META_FUNC(linalg_lu_solve)(const Tensor& LU,
+                                 const Tensor& pivots,
+                                 const Tensor& B,
+                                 bool left,
+                                 bool adjoint) {
+  // dtype
+  at::native::checkFloatingOrComplex(LU, "torch.linalg.lu_solve");
+  TORCH_CHECK(LU.scalar_type() == B.scalar_type(),
+              "linalg.lu_solve: Expected LU and B to have the same dtype, but found LU of type ",
+              LU.scalar_type(), " and B of type ", B.scalar_type(), " instead");
+  TORCH_CHECK(pivots.dtype() == at::kInt,
+              "linalg.lu_solve: pivots should be a Tensor of scalar type torch.int32");
+
+  // matrix shapes
+  at::native::squareCheckInputs(LU, "torch.linalg.lu_solve");
+  at::native::checkInputsSolver(LU, B, left, "linalg.lu_solve");
+  //
+  TORCH_CHECK(LU.size(-1) == pivots.size(-1),
+              "linalg.lu_solve: Number of pivots per batch should be same as the dimension of the matrix");
+
+  // batches
+  TORCH_CHECK(
+      LU.sizes().slice(0, LU.dim() - 1).equals(pivots.sizes()),
+      "linalg.lu_solve: Expected LU.shape[:-1] and pivots.shape to be the same, but got pivots with shape ",
+      pivots.sizes(), " instead");
+
+  // This one checks that B can be broadcasted to the shape of A
+  auto B_broadcast_size = std::get<0>(at::native::_linalg_broadcast_batch_dims(B, LU));
+  auto result_strides = at::native::contiguous_strides(B_broadcast_size, /*column_major=*/left);
+
```

Comment:
https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-fortran/top/lapack-routines/lapack-linear-equation-routines/lapack-linear-equation-computational-routines/solve-sys-of-linear-equations-lapack-computation/getrs.html?wapkw=getrs mentions that the second dimension of both `A` and `B` should be at least 1. Is it checked somewhere?

Reply:
At the beginning of `TORCH_IMPL_FUNC`.

Reply:
Do you mean the leading dimension (`lda`, `ldb`)? The second dimension of `A` (`n`) and `B` (`nrhs`) can be 0.

Reply:
I failed to find that in neither the squre check nor the solver check.. Am I missing something?

Reply:
I think we should be fine as `n` and `nrsh` are `>=0` and the leading dimension is at least 1.

Reply:
This is the line of code that ensure max(1, ...): https://github.com/pytorch/pytorch/blob/3d4e60e4fdcb2ab611d65dc5eff4c110880c7137/aten/src/ATen/native/BatchLinearAlgebraKernel.cpp#L1048